weather_dl/download_pipeline/clients.py

```diff
@@ -152,6 +170,90 @@ def __exit__(self, exc_type, exc_value, traceback):
         self._redirector.__exit__(exc_type, exc_value, traceback)
 
 
+class APIRequestExtended(api.APIRequest):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def fetch(self, request):
+        status = None
+
+        self.connection.submit("%s/%s/requests" % (self.url, self.service), request)
+        self.log("Request submitted")
+        self.log("Request id: " + self.connection.last.get("name"))
+        if self.connection.status != status:
+            status = self.connection.status
+            self.log("Request is %s" % (status,))
+
+        while not self.connection.ready():
+            if self.connection.status != status:
+                status = self.connection.status
+                self.log("Request is %s" % (status,))
+            self.connection.wait()
+
+        if self.connection.status != status:
+            status = self.connection.status
+            self.log("Request is %s" % (status,))
+
+        result = self.connection.result()
+        return result
+
+    def download(self, result, target=None):
+        if target:
+            if os.path.exists(target):
+                # Empty the target file, if it already exists, otherwise the
+                # transfer below might be fooled into thinking we're resuming
+                # an interrupted download.
+                open(target, "w").close()
+
+            size = -1
+            tries = 0
```

Comment:
Following up with the comment below: We could use Beam's retry logic with exponential backoff (via the decorator) instead of re-using ECMWF's implementation. WDYT?

Reply:
Yes, necessary code changes have been done.